stock_picking_send_by_mail/data/stock_picking_mail.xml

```diff
@@ -26,7 +20,7 @@
     <p style="border-left: 1px solid #8e0000; margin-left: 30px;">
        &nbsp;&nbsp;<strong>REFERENCES</strong><br />
        &nbsp;&nbsp;Picking number: <strong>${object.name}</strong><br />
-       &nbsp;&nbsp;Picking date: ${format_datetime(object.date, tz=user.tz, lang_code=object.partner_id.lang)} <br />
+       &nbsp;&nbsp;Picking date: ${format_datetime(object.date, tz=user.tz)} <br />
```

Comment:
You are removing lang_code=object.partner_id.lang when in the translation files keeps there